### PR TITLE
refactor: use react server components for Puck

### PR DESCRIPTION
**Pull request by chrisvxd**
Implementing the canary react server components feature. Partly for testing, but no harm in us having it.

### CODE REVIEW

File: app/layout.tsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import "@measured/puck/dist/index.css";
+import "@measured/puck/puck.css";
```
Comment (monospaced) [inline app/layout.tsx:1]: Nice!

File: package.json [hunk 2]
```diff
@@ -3,14 +3,14 @@
   "version": "0.0.2",
   "private": true,
   "dependencies": {
-    "@measured/puck": "^0.9.0",
-    "@measured/puck-plugin-heading-analyzer": "^0.9.0",
+    "@measured/puck": "0.12.0-canary.53e3fc5",
+    "@measured/puck-plugin-heading-analyzer": "0.12.0-canary.53e3fc5",
```
Comment (monospaced) [inline package.json:7]: We should remember to hop back on mainline one these features have been released

**Review by monospaced — APPROVED** [review]
RSC, whatever Next! 🥁

File: package.json [hunk 3]
```diff
@@ -3,14 +3,14 @@
   "version": "0.0.2",
   "private": true,
   "dependencies": {
-    "@measured/puck": "^0.9.0",
-    "@measured/puck-plugin-heading-analyzer": "^0.9.0",
+    "@measured/puck": "0.12.0-canary.c066397",
+    "@measured/puck-plugin-heading-analyzer": "0.12.0-canary.c066397",
```
Comment (monospaced) [inline package.json:7]: This could be set back to `"^0.12.0"` now

Comment (chrisvxd) [reply package.json]: Done!

**Pull request merged**